pandas/core/arrays/integer.py

```diff
@@ -316,13 +316,13 @@ def __init__(self, values: np.ndarray, mask: np.ndarray, copy: bool = False):
         super().__init__(values, mask, copy=copy)
 
     def __neg__(self):
-        return type(self)(-self._data, self._mask)
+        return type(self)(-self._data, self._mask.copy())
 
     def __pos__(self):
         return self
```

Comment:
We should do the same for `__pos__`?

Reply:
probably not for a backport. but indeed this needs some discussion to ensure consistency across the codebase. see TimedeltaArray

```
    def __pos__(self) -> TimedeltaArray:
        return type(self)(self._data, freq=self.freq)
```

```
>>> import pandas as pd
>>> pd.__version__
'1.3.0.dev0+804.g3289f82975'
>>>
>>> arr = pd.timedelta_range(0, periods=10)._values
>>> arr
<TimedeltaArray>
['0 days', '1 days', '2 days', '3 days', '4 days', '5 days', '6 days',
 '7 days', '8 days', '9 days']
Length: 10, dtype: timedelta64[ns]
>>>
>>> arr2 = +arr
>>>
>>> arr2 is arr
False
>>>
>>> arr2._data is arr._data
True
>>>
>>> arr2[5] = None
>>>
>>> arr
<TimedeltaArray>
['0 days', '1 days', '2 days', '3 days', '4 days',      NaT, '6 days',
 '7 days', '8 days', '9 days']
Length: 10, dtype: timedelta64[ns]
>>>
>>>
```

Reply:
OK, I see, so that's indeed something we should fix more generally. Is there already an issue about it?

Reply:
In the context here I had been thinking that something like `s1 = +s` should behave the same way as `s1 = s`, but actually I think you are right. Likely `+s` should return something new instead of simply a reference to the thing.

Reply:
> Is there already an issue about it?

no. will need to create one, since the issue that this PR addresses is slightly different. #39943 is about different arrays sharing a mask, whereas returning self for \_\_pos__, although maybe incorrect, does not create the inconsistencies when assigning null and non-null vales